blog/views.py

```diff
@@ -0,0 +1,96 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.shortcuts import render, redirect
+from django.urls import reverse_lazy, reverse
+from django.views import generic
+from django.views.generic.edit import FormMixin
+
+from blog.forms import CommentaryForm
+from blog.models import Post, Commentary
+
+
+class IndexListView(generic.ListView):
+    model = Post
+    queryset = Post.objects.select_related("owner")
```

Comment:
You should provide either model or queryset, in this exact class queryset is better.